DOC: update the pandas.DataFrame.any docstring

smusali opened the pull request:
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [+] PR title is "DOC: update the <your-function-or-method> docstring"
- [+] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [+] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [+] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [+] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
####################### Docstring (pandas.DataFrame.any) #######################
################################################################################

Return whether any element is True over requested axis.

Unlike :meth:`DataFrame.all`, this performs an *or* operation. If any of the
values along the specified axis is True, this will return True.

Parameters
----------
axis : int, default 0
    Select the axis which can be 0 for indices and 1 for columns.
skipna : boolean, default True
    Exclude NA/null values. If an entire row/column is NA, the result
    will be NA.
level : int or level name, default None
    If the axis is a MultiIndex (hierarchical), count along a
    particular level, collapsing into a Series.
bool_only : boolean, default None
    Include only boolean columns. If None, will attempt to use everything,
    then use only boolean data. Not implemented for Series.
**kwargs : any, default None
    Additional keywords have no affect but might be accepted for
    compatibility with numpy.

Returns
-------
any : Series or DataFrame (if level specified)

See Also
--------
pandas.DataFrame.all : Return whether all elements are True.

Examples
--------
**Series**

For Series input, the output is a scalar indicating whether any element
is True.

>>> pd.Series([True, False]).any()
True

**DataFrame**

Whether each column contains at least one True element (the default).

>>> pd.DataFrame({
...     "A": [1, 2, 3],
...     "B": [4, 5, 6]
... }).any()
A    True
B    True
dtype: bool

Aggregating over the columns.

>>> pd.DataFrame({
...     "A": [True, False, True],
...     "B": [4, 5, 6]
... }).any(axis='columns')
0    True
1    True
2    True
dtype: bool

>>> pd.DataFrame({
...     "A": [True, False, True],
...     "B": [4, 0, 6]
... }).any(axis='columns')
0    True
1    False
2    True
dtype: bool

`any` for an empty DataFrame is an empty Series.

>>> pd.DataFrame([]).any()
Series([], dtype: bool)

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Use only one blank line to separate sections or paragraphs
	Errors in parameters section
		Parameters {'kwargs'} not documented
		Unknown parameters {'**kwargs'}
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.


Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

smusali committed:
pandas.DataFrame.any

smusali committed:
Updated pandas.core.generic.py...

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7825,6 +7828,49 @@ def _doc_parms(cls):
+_any_also = """\
+See Also
+--------
+pandas.DataFrame.all : Return whether all elements are True \
```
I don't think you need the trailing `\` here do you?

TomAugspurger replied:
to be clear, you do need the one on the first line, jsut not these.

jorisvandenbossche replied:
yes, and the same for the ones below as well

TomAugspurger replied:
We want them, else we get long lines in the text docstring liek

```
One dimensional boolean pandas.Series is returned. Unlike pandas.DataFrame.all, pandas.DataFrame.any performs OR operation; in other word, if any of the values along the specified axis is True, pandas.DataFrame.any will return True.

```

smusali replied:
@TomAugspurger & @jorisvandenbossche , let's say I removed `\` and , then the result of `git diff origin/master -u -- "*.py" | flake8 --diff` is going to be `pandas/core/generic.py:7834:80: E501 line too long (83 > 79 characters)`, since the line is `pandas.DataFrame.all : Return whether all elements are True over requested axis.` - I don't want to break the line with `\n`, instead, I'm using `\`. There is exactly same reason behind the cases I used `\`.

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7825,6 +7828,49 @@ def _doc_parms(cls):
+_any_examples = """\
+Examples
+--------
+By default, any from an empty DataFrame is empty Series::
```
No double colon, just a `.`

smusali replied:
@TomAugspurger according to this [documentation](http://www.sphinx-doc.org/en/stable/markup/code.html), double colon is required to show code samples.

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7825,6 +7828,49 @@ def _doc_parms(cls):
+--------
+By default, any from an empty DataFrame is empty Series::
+
+    >> pd.DataFrame([]).any()
```
Three `>`. Doesn't need to be indented.

smusali replied:
@TomAugspurger , you mean, without double-colon and three `>`, is it going to show code samples as required?

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7825,6 +7828,49 @@ def _doc_parms(cls):
+
+Non-boolean values will always give True::
+
+    >> pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]}).any()
```
same.

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7825,6 +7828,49 @@ def _doc_parms(cls):
+
+It is performing OR along the specified axis::
+
+    >> pd.DataFrame({"A": [1, False, 3], "B": [4, 5, 6]}).any(axis=1)
```
same

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7825,6 +7828,49 @@ def _doc_parms(cls):
+    2    True
+    dtype: bool
+
+    >> pd.DataFrame({"A": [1, False, 3], "B": [4, False, 6]}).any(axis=1)
```
same.

TomAugspurger reviewed (changes requested):
Hmm looks like some extra files were committed. I thikn we have another PR adding `savefig` to our gitignore.

Can you remove those files `rm -rf doc/source/savefig` and then update your PR. I thikn with `git rm doc/source/savefig`.

Reviewers: we should wait for at least one CI to finish since this is changing parameters passed through the functions making the docstrings.

jorisvandenbossche commented on pandas/core/generic.py:
```diff
@@ -7825,6 +7828,49 @@ def _doc_parms(cls):
+_any_desc = """\
+Return whether any element is True over requested axis.
+
+One dimensional pandas.Series having boolean values will be returned. \
```
pandas.Series -> Series

jorisvandenbossche replied:
I would also say "boolean Series" instead of "Series having boolean values"

jorisvandenbossche commented on pandas/core/generic.py:
```diff
@@ -7825,6 +7828,49 @@ def _doc_parms(cls):
+One dimensional pandas.Series having boolean values will be returned. \
+Unlike pandas.DataFrame.all, pandas.DataFrame.any performs OR operation; \
+in other word, if any of the values along the specified axis is True, \
+pandas.DataFrame.any will return True."""
```
Can you also mention here something that for Series the return value is a single boolean value?

jorisvandenbossche reviewed (commented):
Can you also add an example with Series ? (the docstring is shared for both Series and DataFrame)

jreback commented:
this needs a rebase now

smusali committed:
Removing extra files...

smusali committed:
Requested changes...

TomAugspurger committed:
Merge remote-tracking branch 'upstream/master' into smusali-booldoc

smusali committed:
Little update...

TomAugspurger commented:
@smusali I'm doing the rebase / merge. 1 minute.

TomAugspurger committed:
Updated

TomAugspurger committed:
Merge branch 'booldoc' of https://github.com/smusali/pandas into smusali-booldoc

TomAugspurger commented:
@smusali fixed the merge conflict. Also made an update to the grammar and examples.

I changed the examples to have consistent types for the columns. In general, having a mix like `[4, False, 6]` is less common than having all bools or all ints like`[4, 0, 6]`

smusali committed:
Added one more example related to Series and improved long desc...

smusali committed:
Fixed conflict and udpated long_desc and example: added Series example...

smusali committed:
Fixed some issues...

smusali commented:
Done some requested changes and made some fixes - please, review, @TomAugspurger, @jreback  and @jorisvandenbossche; thanks in advance!

TomAugspurger committed:
Updated

Split Series and DataFrame

Edgecase last in examples.

Use axis='columns'

Simplify extended description.

TomAugspurger commented:
```
################################################################################
####################### Docstring (pandas.DataFrame.any) #######################
################################################################################

Return whether any element is True over requested axis.

Unlike :meth:`DataFrame.all`, this performs an *or* operation. If any of the
values along the specified axis is True, this will return True.

Parameters
----------
axis : int, default 0
    Select the axis which can be 0 for indices and 1 for columns.
skipna : boolean, default True
    Exclude NA/null values. If an entire row/column is NA, the result
    will be NA.
level : int or level name, default None
    If the axis is a MultiIndex (hierarchical), count along a
    particular level, collapsing into a Series.
bool_only : boolean, default None
    Include only boolean columns. If None, will attempt to use everything,
    then use only boolean data. Not implemented for Series.
**kwargs : any, default None
    Additional keywords have no affect but might be accepted for
    compatibility with numpy.

Returns
-------
any : Series or DataFrame (if level specified)

See Also
--------
pandas.DataFrame.all : Return whether all elements are True.

Examples
--------
**Series**

For Series input, the output is a scalar indicating whether any element
is True.

>>> pd.Series([True, False]).any()
True

**DataFrame**

Whether each column contains at least one True element (the default).

>>> pd.DataFrame({
...     "A": [1, 2, 3],
...     "B": [4, 5, 6]
... }).any()
A    True
B    True
dtype: bool

Aggregating over the columns.

>>> pd.DataFrame({
...     "A": [True, False, True],
...     "B": [4, 5, 6]
... }).any(axis='columns')
0    True
1    True
2    True
dtype: bool

>>> pd.DataFrame({
...     "A": [True, False, True],
...     "B": [4, 0, 6]
... }).any(axis='columns')
0    True
1    False
2    True
dtype: bool

`any` for an empty DataFrame is an empty Series.

>>> pd.DataFrame([]).any()
Series([], dtype: bool)

################################################################################
################################## Validation ##################################
################################################################################
```

smusali commented:
@TomAugspurger , do u have any more change request?

TomAugspurger committed:
Explicit examples

TomAugspurger commented:
Just updated the examples a tad to show the dataframes.
